eslint/plugins/header/header.cjs

```diff
@@ -0,0 +1,272 @@
+    if (all[0].type.toLowerCase() === "block") {
+        return [all[0]];
+    }
+    for (var i = 1; i < all.length; ++i) {
+        var txt = context.getSourceCode().getText().slice(all[i - 1].range[1], all[i].range[0]);
+        if (!txt.match(/^(\r\n|\r|\n)$/)) {
+            break;
+        }
+    }
+    return all.slice(0, i);
+}
+
+function genCommentBody(commentType, textArray, eol, numNewlines) {
+    var eols = eol.repeat(numNewlines);
+    if (commentType === "block") {
+        return "/*" + textArray.join(eol) + "*/" + eols;
+    } else {
+        return "//" + textArray.join(eol + "//") + eols;
+    }
+}
+
+function genCommentsRange(context, comments, eol) {
+    var start = comments[0].range[0];
+    var end = comments.slice(-1)[0].range[1];
+    if (context.getSourceCode().text[end] === eol) {
+        end += eol.length;
+    }
+    return [start, end];
+}
+
+function genPrependFixer(commentType, node, headerLines, eol, numNewlines) {
+    return function(fixer) {
+        return fixer.insertTextBefore(
+            node,
+            genCommentBody(commentType, headerLines, eol, numNewlines)
+        );
+    };
+}
+
+function genReplaceFixer(commentType, context, leadingComments, headerLines, eol, numNewlines) {
+    return function(fixer) {
+        return fixer.replaceTextRange(
+            genCommentsRange(context, leadingComments, eol),
+            genCommentBody(commentType, headerLines, eol, numNewlines)
+        );
+    };
+}
+
+function findSettings(options) {
+    var lastOption = options.length > 0 ? options[options.length - 1] : null;
+    if (typeof lastOption === "object" && !Array.isArray(lastOption) && lastOption !== null
+        && !Object.prototype.hasOwnProperty.call(lastOption, "pattern")) {
+        return lastOption;
+    }
+    return null;
+}
+
+function getEOL(options) {
+    var settings = findSettings(options);
+    if (settings && settings.lineEndings === "unix") {
+        return "\n";
+    }
+    if (settings && settings.lineEndings === "windows") {
+        return "\r\n";
+    }
+    return os.EOL;
+}
+
+function hasHeader(src) {
+    if (src.substr(0, 2) === "#!") {
+        var m = src.match(/(\r\n|\r|\n)/);
+        if (m) {
+            src = src.slice(m.index + m[0].length);
+        }
+    }
+    return src.substr(0, 2) === "/*" || src.substr(0, 2) === "//";
+}
+
+function matchesLineEndings(src, num) {
+    for (var j = 0; j < num; ++j) {
+        var m = src.match(/^(\r\n|\r|\n)/);
+        if (m) {
+            src = src.slice(m.index + m[0].length);
+        } else {
+            return false;
+        }
+    }
+    return true;
+}
+
+module.exports = {
+    meta: {
+        type: "layout",
+        fixable: "whitespace",
+        schema: [
+            {
+                enum: ["block", "line"]
+            },
+            {
+                type: "array",
+            }
+        ]
+    },
```

Comment:
official `eslint-plugin-header` is missing this `schema` definition, which will make eslint v9 fail whenever there's more than 0 options.